From 150a83c90f03e644dc354df2ee60717770e6243c Mon Sep 17 00:00:00 2001
From: Ivan Mironov <mironov.ivan@gmail.com>
Date: Wed, 17 Sep 2025 08:45:35 +0200
Subject: [PATCH] Add application ID

---
 puffin_viewer/src/main.rs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/puffin_viewer/src/main.rs b/puffin_viewer/src/main.rs
index 2efbf011bc49..f0f22d2ab77a 100644
--- a/puffin_viewer/src/main.rs
+++ b/puffin_viewer/src/main.rs
@@ -62,7 +62,8 @@ fn main() -> Result<(), eframe::Error> {
             .with_app_id("puffin_viewer")
             .with_drag_and_drop(true)
             .with_inner_size([1400.0, 1000.0])
-            .with_icon(icon),
+            .with_icon(icon)
+            .with_app_id("com.github.EmbarkStudios.puffin.puffin_viewer"),
         ..Default::default()
     };
 
-- 
2.51.0

